app/src/main/kotlin/compiler/lowlevel/Instruction.kt

```
@@ -19,12 +19,25 @@ sealed class Instruction : Asmable {
         }
     }
 
-    override fun writeAsm(output: PrintWriter, registers: Map<Register, Register>) {
-        TODO() // Remove this and implement toAsm in each Instruction instance separately
-    }
+    abstract fun toAsm(registers: Map<Register, Register>): String
+
+    override fun writeAsm(output: PrintWriter, registers: Map<Register, Register>) = output.write(toAsm(registers))
 
     sealed class ConditionalJumpInstruction : Instruction() {
         abstract val targetLabel: String
+
+        override fun toAsm(registers: Map<Register, Register>) = when (this) {
+            is JmpEq -> "je"
+            is JmpNEq -> "jne"
+            is JmpLt -> "jl"
+            is JmpLtEq -> "jle"
+            is JmpGt -> "jg"
+            is JmpGtEq -> "jge"
+            is JmpZ -> "jz"
+            is JmpNZ -> "jnz"
+            is Dummy -> "dummy-conditional-jump"
```

We should throw an exception here.